### PR TITLE
Allow the specification of server group policies when provisioning openstack

### PR DESCRIPTION
**Pull request by tzumainn**
#### What does this PR do?
Allow the specification of master and infra server group policies when provisioning openstack

#### How should this be manually tested?

Uncomment the following parameters in all.yml:

    # Specify server group policies for master and infra nodes. Nova must be configured to
    # enable these policies.
    openstack_master_server_group_policies: [affinity]
    openstack_infra_server_group_policies: [affinity]

You can specify 'affinity' or 'anti-affinity'. Run OpenStack provisioning; it should complete with no errors.

#### Is there a relevant Issue open for this?
N/A

#### Who would you like to review this?
cc: @tomassedovic @bogdando @Tlacenka PTAL

### CODE REVIEW

File: roles/openstack-stack/templates/heat_stack_server.yaml.j2 [hunk 1]
```diff
@@ -154,6 +159,7 @@ resources:
         host-type: { get_param: type }
         sub-host-type:    { get_param: subtype }
         node_labels: { get_param: node_labels }
+      scheduler_hints: { get_param: scheduler_hints }
```
Comment (bogdando) [inline roles/openstack-stack/templates/heat_stack_server.yaml.j2]: indentation fix?

Comment (tzumainn) [reply roles/openstack-stack/templates/heat_stack_server.yaml.j2]: @bogdando thanks for pointing that out! It's actually correct; node_labels is part of the metadata entry (you can see it if you look at the full file).

File: playbooks/provisioning/openstack/README.md [hunk 2]
```diff
@@ -308,6 +308,18 @@ In order to set a custom entrypoint, update `openshift_master_cluster_public_hos
...
+
+### Specifying server group policies
+
+You can specify server group policies for infra and master nodes using the following
+parameters in `inventory/group_vars/all.yml`:
+
+    ## Specify server group policies for master and infra nodes. Nova must be configured to
+    ## enable these policies.
+    #openstack_master_server_group_policies: [affinity]
+    #openstack_infra_server_group_policies: [affinity]
```
[the remark this answers is not in the document]
Comment (tzumainn) [reply playbooks/provisioning/openstack/README.md]: Thanks! I've added the requested link. I'd prefer to deail with node filters in a different PR; let me know if you disagree!

**Comment by tzumainn** [comment]
@tomassedovic makes sense! Updated.

**Pull request merged**